libs/core/async_sycl/tests/unit/sycl_vector_add_executor.cpp

```diff
@@ -0,0 +1,181 @@
+
+    // Test post and get_future methods
+    hpx::sycl::experimental::sycl_executor exec(
+        cl::sycl::default_selector{});
+    std::cout << "Running on device: "
+              << exec.get_device().get_info<cl::sycl::info::device::name>()
+              << std::endl;
+    // Launch kernel one-way
+    exec.post(&cl::sycl::queue::submit, [&](cl::sycl::handler& h) {
+        cl::sycl::accessor a(a_buf, h, cl::sycl::read_only);
+        cl::sycl::accessor b(b_buf, h, cl::sycl::read_only);
+        cl::sycl::accessor add(
+            add_buf, h, cl::sycl::write_only, cl::sycl::no_init);
+        h.parallel_for(num_items, [=](auto i) { add[i] = a[i] + b[i]; });
+    });
+    // NOTE: This is NOT the recommended way to get a future for a kernel
+    // launch as exec.get_future needs to submit an internal dummy kernel
+    // to get an event to create this future!
+    //
+    // It is instead recommended to use exec.async_execute to get a future!
+    // exec.get_future is merely a convenience method in case we get a
+    // command_queue from some third-party library and need a future for
+    // the current point the queue
+    auto my_manual_fut = exec.get_future();
+    if (my_manual_fut.is_ready())
+    {
+        std::cerr << "ERROR: Manual get_future using internal dummy kernel "
+                     "is immediately ready "
+                  << "(thus probably not asynchronous at at all)!"
+                  << std::endl;
+        std::terminate();
+    }
+    else
+    {
+        std::cout << "OKAY: Manual get_future hpx::future is NOT ready "
+                     "immediately "
+                     "after launch!"
+                  << std::endl;
+    }
+    my_manual_fut.get();
+}
+
+int hpx_main(int, char**)
+{
+    static_assert(vector_size >= 6, "vector_size unreasonably small");
+    // Enable polling for the future
+    hpx::sycl::experimental::detail::register_polling(hpx::resource::get_thread_pool(0));
+    std::cout << "SYCL Future polling enabled!\n";
+    std::cout << "SYCL language version: " << SYCL_LANGUAGE_VERSION << "\n";
+    // Input vectors
+    std::vector<size_t> a(vector_size), b(vector_size), add_parallel(vector_size);
+
+    // Run tests
+    std::cout << "Test async execute:" << std::endl;
+    fill_vector_add_input(a, b, add_parallel);
+    VectorAdd_test1(a, b, add_parallel);
+    check_vector_add_results(a, b, add_parallel); 
+    print_vector_results(a, b, add_parallel);
+    std::cout << "Test post and get_future() member method:" << std::endl;
+    fill_vector_add_input(a, b, add_parallel); // also resets result buffer
+    VectorAdd_test1(a, b, add_parallel);
+    check_vector_add_results(a, b, add_parallel);
+    print_vector_results(a, b, add_parallel);
+
+    // Cleanup
+    std::cout << "Disabling SYCL future polling.\n";
+    hpx::sycl::experimental::detail::unregister_polling(hpx::resource::get_thread_pool(0));
+    return hpx::finalize();
+}
+
+int main(int argc, char* argv[])
```

Comment:
Shouldn't there be an empty main() implementation in case `HPX_HAVE_SYCL` is not defined?

Reply:
Hmm, good question. Currently, the tests are not being built when HPX_WITH_SYCL=OFF (and thus HPX_HAVE_SYCL is undefined) -- this makes the entire HPX_HAVE_SYCL ifdef in there superfluous. 

That being said, I put it in there in case someone just wanted to quickly look up how to use HPX with SYCL (and that one is good to know). Since I don't really want to remove it for that reason (at least until we have some more examples, but that's for a different PR), I went ahead and added a main implementation for when HPX_HAVE_SYCL is not defined to the unit tests (making it more complete as an example). This also serves the second purpose of checking that HPX_HAVE_SYCL got defined correctly (as those dummy mains return 1 and thus fail the test). Added in f4844b513b61903dc32b436413ebee0e0b551e59

Reply:
> Hmm, good question. Currently, the tests are not being built when HPX_WITH_SYCL=OFF (and thus HPX_HAVE_SYCL is undefined) -- this makes the entire HPX_HAVE_SYCL ifdef in there superfluous.

Yes, that was what I was wondering about... Either always compile and run the test (in which case you need the empty `main()`) or remove the HPX_HAVE_SYCL ifdef.